docs/tutorial/deploy.rst

```diff
@@ -0,0 +1,120 @@
+    pip install wheel
+
+Running ``setup.py`` with Python gives you a command line tool to issue
+build-related commands. The ``bdist_wheel`` command will build a wheel
+distribution file.
+
+.. code-block:: none
+
+    python setup.py bdist_wheel
+
+You can find the file in ``dist/flaskr-1.0.0-py3-none-any.whl``. The
+file name is the name of the project, the version, and some tags about
+the file can install.
+
+Copy this file to another machine,
+:ref:`set up a new virtualenv <install-create-env>`, then install the
+file with ``pip``.
+
+.. code-block:: none
+
+    pip install flaskr-1.0.0-py3-none-any.whl
+
+Pip will install your project along with its dependencies.
+
+Since this is a different machine, you need to run ``init-db`` again to
+create the database in the instance folder.
+
+.. code-block:: none
+
+    export FLASK_APP=flaskr
+    flask init-db
+
+When Flask detects that it's installed (not in editable mode), it uses
+a different directory for the instance folder. You can find it at
+``venv/var/flaskr-instance`` instead.
+
+
+Configure the Secret Key
+------------------------
+
+In the beginning of the tutorial that you gave a default value for
+:data:`SECRET_KEY`. This should be changed to some random bytes in
+production. Otherwise, attackers could use the public ``'dev'`` key to
+modify the session cookie, or anything else that uses the secret key.
+
+You can use the following command to output a random secret key:
+
+.. code-block:: none
+
+    python -c 'import os; print(os.urandom(16))'
+
+    b'_5#y2L"F4Q8z\n\xec]/'
+
+Create the ``config.py`` file in the instance folder, which the factory
+will read from if it exists. Copy the generated value into it.
+
+.. code-block:: python
+    :caption: ``venv/var/flaskr-instance/config.py``
+
+    SECRET_KEY = b'_5#y2L"F4Q8z\n\xec]/'
+
+You can also set any other necessary configuration here, although
+``SECRET_KEY`` is the only one needed for Flaskr.
+
+
+Run with a Production Server
+----------------------------
+
+When running publicly rather than in development, you should not use the
+built-in development server (``flask run``). The development server is
+provided by Werkzeug for convenience, but is not designed to be
+particularly efficient, stable, or secure.
+
+Instead, use a production WSGI server. For example, to use `Waitress`_,
+first install it in the virtual environment:
+
+.. code-block:: none
+
+    pip install waitress
+
+You need to tell Waitress about your application, but it doesn't use
+``FLASK_APP`` like ``flask run`` does. You need to tell it to import and
+call the application factory to get an application object.
+
+.. code-block:: none
+
+    waitress-serve --call 'flaskr:create_app'
+
+    Serving on http://0.0.0.0:8080
+
+See :doc:`/deploying/index` for a list of many different ways to host
+your application. Waitress is just an example, you may choose a
+different setup for your project.
+
+.. _Waitress: https://docs.pylonsproject.org/projects/waitress/
```

Comment:
It is better to mention other production ready servers like gunicorn

Reply:
Waitress advertises itself as production ready, and works on both Windows and Linux. Gunicorn only works on Linux. Nothing against Gunicorn, I just wanted something that would work for everyone in the tutorial.

Reply:
I think I link to the deploy docs at the top, I could move that link down here.

Reply:
I think just the note that "you may choose a different setup" is probably the right choice. If you start naming servers then either you're endorsing the ones that you name, or you've got a list of *all* the servers. If anything, "Waitress is an example, advertised as production ready and works on Windows and Linux. You may choose..." would be the only change I could see making, but I don't even think it's necessary.

Pylons is a pretty venerable project - I trust that their product is worthwhile. If someone has no preference or experience, I don't think pointing them towards Pylons is a bad idea. If they do have a preference then they already should know how to (figure out how to) setup their preference. And the fact that it's using wsgi means it should be simple to replace waitress with whatever else.